libtenzir/builtins/operators/serve.cpp

```diff
@@ -486,21 +485,24 @@ struct serve_manager_state {
     if (delivered) {
       return found->get_rp;
     }
+    found->delayed_attempt.dispose();
     found->delayed_attempt = detail::weak_run_delayed(
       self, request.limits.timeout,
-      [this, continuation_token = request.continuation_token]() mutable {
+      [this, serve_id = request.serve_id,
+       continuation_token = request.continuation_token]() mutable {
         const auto found
           = std::find_if(ops.begin(), ops.end(), [&](const auto& op) {
-              return op.continuation_token == continuation_token;
+              return op.serve_id == serve_id;
             });
         if (found == ops.end()) {
           TENZIR_DEBUG("unable to find serve request after timeout expired");
           return;
         }
+        TENZIR_ASSERT(not found->done);
+        TENZIR_ASSERT(found->continuation_token == continuation_token);
```

Comment:
Wouldn't this trigger if a user successfully retries the `/serve` request before the timeout has run out?

Reply:
No, in that case the delayed run would be disposed.